tree/ntuple/v7/src/RDaos.cxx

```diff
@@ -85,7 +82,10 @@ ROOT::Experimental::Detail::RDaosObject::RDaosObject(RDaosContainer &container,
                                                      ObjClassId cid)
 {
    if (!cid.IsUnknown())
-      daos_obj_generate_id(&oid, DAOS_OF_DKEY_UINT64 | DAOS_OF_AKEY_UINT64 /*| DAOS_OF_ARRAY_BYTE*/, cid.fCid, 0);
+      daos_obj_generate_oid(container.fContainerHandle, &oid,
+                            static_cast<daos_otype_t>(DAOS_OT_DKEY_UINT64 | DAOS_OT_AKEY_UINT64), cid.fCid,
+                            DAOS_OCH_RDD_DEF, 0);
```

Comment:
We may also explicitly mention a sharding OC hint flag (although as per current DAOS version, it behaves the same if left unspecified).
```suggestion
                            DAOS_OCH_RDD_DEF | DAOS_OCH_SHD_DEF, 0);
```